lib/actionpack/all/actionpack.rbi

```diff
@@ -33,35 +33,203 @@ class ActionDispatch::Response
+  def merge(other_hash); end
+
+  sig { returns(T.untyped) }
+  def parameters; end
+
   sig { returns(T.self_type) }
   def permit!; end
 
   # You can pass _a lot_ of stuff to permit, so filters is left untyped for now.
   sig { params(filters: T.untyped).returns(ActionController::Parameters) }
   def permit(*filters); end
 
+  sig { params(new_permitted: T.untyped).void }
+  def permitted=(new_permitted); end
+
   sig { returns(T::Boolean) }
   def permitted?; end
 
+  sig { params(block: T.untyped).returns(T.untyped) }
+  def reject!(&block); end
+
+  # delete_if is an alias of reject!
+  sig { params(block: T.untyped).returns(T.untyped) }
+  def delete_if(&block); end
+
+  sig { params(block: T.untyped).returns(T.untyped) }
+  def reject(&block); end
+
   sig { params(key: T.any(Symbol, T::Array[Symbol])).returns(ActionController::Parameters) }
   def require(key); end
 
   # required is an alias of require
   sig { params(key: T.any(Symbol, T::Array[Symbol])).returns(ActionController::Parameters) }
   def required(key); end
 
-  sig { returns(ActiveSupport::HashWithIndifferentAccess) }
+  sig { params(other_hash: T.untyped).returns(ActionController::Parameters) }
+  def reverse_merge!(other_hash); end
+
+  # with_defaults! is an alias of reverse_merge!
+  sig { params(other_hash: T.untyped).returns(ActionController::Parameters) }
+  def with_defaults!(other_hash); end
+
+  sig { params(other_hash: T.untyped).returns(ActionController::Parameters) }
+  def reverse_merge(other_hash); end
+
+  # with_defaults is an alias of reverse_merge
+  sig { params(other_hash: T.untyped).returns(ActionController::Parameters) }
+  def with_defaults(other_hash); end
+
+  sig { params(block: T.untyped).returns(T.nilable(ActionController::Parameters)) }
+  def select!(&block); end
+  
+  # keep_if is an alias of select!
+  sig { params(block: T.untyped).returns(T.nilable(ActionController::Parameters)) }
+  def keep_if(&block); end
+
+  sig { params(block: T.untyped).returns(ActionController::Parameters) }
+  def select(&block); end
+
+  sig { returns(T.any(Symbol, T::Boolean)) }
+  def self.action_on_unpermitted_parameters; end
+
+  sig { params(obj: T.any(Symbol, T::Boolean)).void }
+  def self.action_on_unpermitted_parameters=(obj); end
+
+  sig { returns(T::Array[T.any(String, Symbol)]) }
+  def self.always_permitted_parameters; end
+
+  sig { params(obj: T::Array[T.any(String, Symbol)]).void }
+  def self.always_permitted_parameters=(obj); end
+
+  sig { returns(T::Boolean) }
+  def self.permit_all_parameters; end
+
+  sig { params(obj: T::Boolean).void }
+  def self.permit_all_parameters=(obj); end
+
+  sig { params(keys: T.any(String, Symbol)).returns(ActionController::Parameters) }
+  def slice!(*keys); end
+
+  sig { params(keys: T.any(String, Symbol)).returns(ActionController::Parameters) }
+  def slice(*keys); end
+
+  sig { returns(T.nilable(ActiveSupport::HashWithIndifferentAccess)) }
```

Comment:
@connorshea Do you know when this will return `nil`? Coz making them returning non-nilable object will be very useful for me.
```
=> <ActionController::Parameters {"a"=>1, "b"=>2} permitted: false>
irb(main):006:0> param.permit().to_hash
=> {}

irb(main):007:0> param = ActionController::Parameters.new
=> <ActionController::Parameters {} permitted: false>
irb(main):009:0> param.permit(:a).to_h
=> {}
irb(main):010:0> param.permit(:a).to_hash
=> {}
```

Reply:
I probably made it nilable out of caution, you can remove it.